packages/web-components/fast-components/src/styles/patterns/button.ts

```diff
@@ -38,457 +45,418 @@ import {
+            color: ${neutralForegroundRest};
+            border-radius: calc(${controlCornerRadius} * 1px);
+            fill: currentcolor;
+            cursor: pointer;
+        }
+        :host .control {
+            background: ${neutralFillRest};
+            border: calc(${strokeWidth} * 1px) solid transparent;
+            flex-grow: 1;
+            box-sizing: border-box;
+            display: inline-flex;
+            justify-content: center;
+            align-items: center;
+            padding: 0 calc((10 + (${designUnit} * 2 * ${density})) * 1px);
+            white-space: nowrap;
+            outline: none;
+            text-decoration: none;
+            color: inherit;
+            border-radius: inherit;
+            fill: inherit;
+            cursor: inherit;
+            font-family: inherit;
+        }
+        .control,
+        .end,
+        .start {
+            font: inherit;
+        }
+        .control.icon-only {
+            padding: 0;
+            line-height: 0;
+        }
+        :host .control${interactivitySelector}:hover {
+            background: ${neutralFillHover};
+            color: ${neutralForegroundHover};
+        }
+        :host .control${interactivitySelector}:active {
+            background: ${neutralFillActive};
+            color: ${neutralForegroundActive};
+        }
+        :host .control:${focusVisible} {
+            background: ${neutralFillFocus};
+            border-color: ${focusStrokeOuter} !important;
+            box-shadow: 0 0 0 calc(${focusStrokeWidth} * 1px) ${focusStrokeOuter} inset !important;
+            color: ${neutralForegroundHover};
+        }
+        .control::-moz-focus-inner {
+            border: 0;
+        }
+        .content {
+            pointer-events: none;
+        }
+        .start,
+        .end {
+            display: flex;
+            pointer-events: none;
+        }
+        .start {
+            margin-inline-end: 11px;
+        }
+        .end {
+            margin-inline-start: 11px;
+        }
+  `.withBehaviors(
+        forcedColorsStylesheetBehavior(
+            css`
+              :host .control {
+                  color: ${SystemColors.ButtonText};
+                  fill: currentcolor;
+              }
+              :host .control${interactivitySelector}:hover {
+                  forced-color-adjust: none;
+                  border-color: ${SystemColors.ButtonText};
+                  background: ${SystemColors.Highlight};
+                  color: ${SystemColors.HighlightText};
+              }
+              :host .control:${focusVisible} {
+                  forced-color-adjust: none;
+                  border-color: ${SystemColors.ButtonText} !important;
+                  box-shadow: 0 0 0 calc(${focusStrokeWidth} * 1px) ${SystemColors.ButtonText} !important;
+                  background: ${SystemColors.ButtonFace};
+                  color: ${SystemColors.ButtonText};
+              }
+              :host([href]) .control {
```

Comment:
I couldn't find href used in button.ts or in button.template.ts. Are these shared styles or should all of the :host([href]) styles be in anchor.styles.ts? I get that you didn't put them here, just a broader question. I'll mark this comment as resolved.

Reply:
I think refactoring to move, the selectors with `href`, into anchor style, is a good idea.